src/Microsoft.AspNetCore.Dispatcher/DispatcherMiddleware.cs

```diff
@@ -53,19 +53,21 @@ public DispatcherMiddleware(IOptions<DispatcherOptions> options, ILogger<Dispatc
                     feature.Values = context.Values;
 
                     await context.ShortCircuit(httpContext);
+
+                    _logger.RequestShortCircuited(context);
                     return;
                 }
 
                 if (context.Endpoint != null)
                 {
-                    _logger.LogInformation("Matched endpoint {Endpoint}", context.Endpoint.DisplayName);
-
+                    _logger.EndpointMatched(context.Endpoint);
                     feature.Endpoint = context.Endpoint;
                     feature.Values = context.Values;
 
                     feature.Handler = entry.HandlerFactory.CreateHandler(feature.Endpoint);
                     if (feature.Handler == null)
                     {
+                        _logger.HandlerNotCreated(entry);
                         throw new InvalidOperationException("Couldn't create a handler, that's bad.");
                     }
 
```

Comment:
It would be good to have another log point here saying that the matcher did not match